server/handlers/payment_handler.coffee

```diff
@@ -90,6 +90,18 @@ PaymentHandler = class PaymentHandler extends Handler
     if (not req.user) or req.user.isAnonymous()
       return @sendForbiddenError(res)
 
+    if pathName is 'admin' and req.user?.isAdmin()
+      payment = new Payment()
+      for key, val of req.body
+        if key in ['purchaser', 'recipient']
+          payment.set key, mongoose.Types.ObjectId(val)
+        else
+          payment.set key, val
+      payment.save (err) =>
+        return @sendDatabaseError(res, err) if err
+        return @sendCreated(res, @formatEntity(req, payment))
+      return
+
```

Comment:
Oh no! Couldn't this be a middleware? 😢 For sure should at least have some tests which make sure it works, and that it fails if you're not an admin.

Reply:
Where should this be instead?  This seemed to be the code path used to create payments, which is why I put it here.